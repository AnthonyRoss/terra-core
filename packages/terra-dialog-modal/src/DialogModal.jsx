import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import 'terra-base/lib/baseStyles';
import styles from './DialogModal.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * React node to be placed within the header area of the dialog.
   */
  header: PropTypes.node.isRequired,
  /**
   * React node to be placed within the footer area of the dialog.
   * Content exceeding the available space will be hidden.
   */
  footer: PropTypes.node.isRequired,
  /**
   * The children to be placed within the main content area of the dialog.
   */
  children: PropTypes.node,
  /**
   * Callback function for when the close button is clicked. The close button will not display if this is not set.
   * On small viewports a back button will be displayed instead of a close button.
   */
  onClose: PropTypes.func,
};

const defaultProps = {
  onClose: null,
  children: null,
};

const DialogModal = ({ header, footer, children, onClose, ...customProps }) => {
  const DialogModalClassNames = cx([
    'dialog-modal',
    customProps.className,
  ]);

  return (<div {...customProps} className={DialogModalClassNames}>{name}</div>);
};

DialogModal.propTypes = propTypes;
DialogModal.defaultProps = defaultProps;

export default DialogModal;
