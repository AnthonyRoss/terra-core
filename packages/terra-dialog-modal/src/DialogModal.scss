:local {
  .dialog-modal {
    /* TODO: Add styles */
  }
}
