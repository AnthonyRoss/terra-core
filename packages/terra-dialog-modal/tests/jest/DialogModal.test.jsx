import React from 'react';
import DialogModal from '../../src/DialogModal';

describe('DialogModal', () => {
  const defaultRender = <DialogModal />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.dialog-modal').text()).toEqual('default');
  });

  // Structure Tests
  it('should have the class dialog-modal', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('dialog-modal');
  });
});
