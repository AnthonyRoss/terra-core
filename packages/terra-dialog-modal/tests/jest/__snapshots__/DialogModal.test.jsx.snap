// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogModal should render a default component 1`] = `
<div
  className="dialog-modal"
>
  default
</div>
`;
